src/WebApi/Logging/TimeElapsedDiagnosticsMiddleware.cs

```diff
@@ -17,19 +19,36 @@ public TimeElapsedDiagnosticsMiddleware(RequestDelegate next)
     public async Task Invoke(HttpContext context)
     {
         Stopwatch sw = Stopwatch.StartNew();
-        try
+        string correlationId = GetCorrelationId(context);
+        SetCorrelationId(correlationId,context);
+        
+        using (LogContext.PushProperty("CorrelationId", correlationId))
         {
-            await _next(context);
-            PushPropertyElapsed(sw);
-            Log.Debug("TimeElapsedDiagnosticsMiddleware - OK");
-        }
-        catch (Exception e)
-        {
-            PushPropertyElapsed(sw);
-            Log.Fatal(e,"TimeElapsedDiagnosticsMiddleware - ERROR Message: {Message}", e.Message);
+            try
+            {
+                await _next.Invoke(context);
+                PushPropertyElapsed(sw);
+                Log.Information("TimeElapsedDiagnosticsMiddleware - OK - CorrelationId: {CorrelationId}",correlationId);
+            }
+            catch (Exception e)
+            {
+                PushPropertyElapsed(sw);
+                Log.Fatal(e,"TimeElapsedDiagnosticsMiddleware - ERROR - CorrelationId: {CorrelationId} Message: {Message}", correlationId, e.Message);
+            }
         }
     }
 
+    private static string GetCorrelationId(HttpContext context)
+    {
+        context.Request.Headers.TryGetValue("correlation-id", out StringValues correlationIds);
+        return correlationIds.FirstOrDefault() ?? Guid.NewGuid().ToString();
+    }
+    
+    private static void SetCorrelationId(string correlationId,HttpContext context)
```

Comment:
```suggestion
    private static void SetCorrelationId(string correlationId, HttpContext context)
```

trivial lint

Reply:
Dear @petrspelos 
Thank you for your valuable input.
I can not use Visual Studio. 
Best practice is to set rules in [.editorconfig](https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2022) for other IDEs.
Could you please do that?

Reply:
@PlesnikJakub